### ISSUE TITLE
AdyenComponent Composable disappears on recomposition 

### ISSUE THREAD

**Issue by ronaldvanduren** [Adyen/adyen-android#1399]
**Describe the bug**
After migrating to 5.0.1 I notice that the `AdyenComponent` is not shown after the initial composition of the screen (using Compose). It works fine if the `AdyenComponent` is added and not removed from the view. But if the `AdyenComponent` is removed and added again, the second time it's not shown. Also, if it's used inside a `HorizontalPager` it's not shown at all. 

In 4.13.3 I implemented the `CardView` in Compose without this issue, the view remains always visible. 

This works(v4.13.3)
```kotlin
@Composable
private fun PaymentMethods(
    .......
    pagerState: PagerState,
    viewState: ......
    onCreditCardInputChanged: (CardComponentState) -> Unit
) {
    
    HorizontalPager(state = pagerState) { currentPage ->
        viewState.paymentMethods.elementAt(currentPage).let { paymentMethod ->
            when (paymentMethod) {
                is PaymentMethod.Twint -> TwintContent()
                is PaymentMethod.CreditCard -> CreditCardContent(
                    cardComponent,
                    onCreditCardInputChanged
                )
            }
        }
    }
}

@Composable
internal fun CreditCardContent(
    cardComponent: CardComponent,
    onCreditCardInputChanged: (CardComponentState) -> Unit = { }
) {
    Column(
        Modifier.height(PAYMENT_CONTENT_HEIGHT.dp)
    ) {
        
        val lifecycleOwner = LocalLifecycleOwner.current

        DisposableEffect(cardComponent) {

            cardComponent.observe(lifecycleOwner) { paymentComponentState ->
                onCreditCardInputChanged.invoke(paymentComponentState)
            }

            onDispose {
                cardComponent.removeObservers(lifecycleOwner)
            }
        }

        AndroidView(factory = { context ->
            CardView(context).apply {
                attach(cardComponent, lifecycleOwner)
            }
        }, modifier = Modifier.fillMaxWidth()
        )
    }
}
```

This doesn't work (v5.01)
```kotlin

@Composable
private fun PaymentMethods(
    ........
    pagerState: PagerState,
    viewState: ....
    onCreditCardInputChanged: (CardComponentState) -> Unit
) {
    ......
    HorizontalPager(state = pagerState) { currentPage ->
        viewState.paymentMethods.elementAt(currentPage).let { paymentMethod ->
            when (paymentMethod) {
                is PaymentMethod.Twint -> TwintContent()
                is PaymentMethod.CreditCard -> CreditCardContent(cardComponent)
            }
        }
    }
}

@Composable
internal fun CreditCardContent(
    cardComponent: CardComponent
) {
    Column(
        Modifier.height(PAYMENT_CONTENT_HEIGHT.dp)
    ) {
        
        AdyenComponent(
            component = cardComponent,
            modifier = Modifier.fillMaxWidth()
        )
    }
}
```

This does work (v5.01)
```kotlin

@Composable
private fun PaymentMethods(
    ........
    pagerState: PagerState,
    viewState: ....
    onCreditCardInputChanged: (CardComponentState) -> Unit
) {
        CreditCardContent(cardComponent)
    ......
}

@Composable
internal fun CreditCardContent(
    cardComponent: CardComponent
) {
    Column(
        Modifier.height(PAYMENT_CONTENT_HEIGHT.dp)
    ) {
        
        AdyenComponent(
            component = cardComponent,
            modifier = Modifier.fillMaxWidth()
        )
    }
}
```

**Expected behavior**
Whenever the `AdyenComponent` is added dynamically to the view tree (e.g. via `HorizontalPager`), then the `AdyenComponent` should be visible. 

**Smartphone:**
 - Device: Pixel 6
 - OS: Android 14

**Comment by ronaldvanduren** [comment]
Thanks for the reply! Unfortunately, setting `beyondBoundsPageCount` does not work consistently. It "sometimes" works when the debugger is attached, this with the fact that in 5.x.x the components make use of flows to initialise the views makes me think it might be a concurrency issue. 

The same behaviour is observed when setting the `contentPadding`, when the debugger or the layout inspector is attached, then the component shows more often. 

I do see however that the AdyenComponentView is inflated and that the CardView is attached, it's just that the CardView is not properly inflated (it seems).

**AdyenComponentView**
![Screenshot 2023-12-01 at 11 47 30](https://github.com/Adyen/adyen-android/assets/4056935/f0ab3b94-39fd-479c-ac1c-b2b7a6eed3b9)

**CardView**
![Screenshot 2023-12-01 at 11 47 44](https://github.com/Adyen/adyen-android/assets/4056935/81fe083c-a87c-43c1-a186-833a6abfa7ec)

**Comment by araratthehero** [comment]
Hi @ronaldvanduren, the fix for this bug has been merged and will be available in our next release.
We will keep you updated when it goes live.

Thanks again for reporting it!

**Comment by araratthehero** [comment]
Hi @ronaldvanduren, we just released [5.1.0](https://github.com/Adyen/adyen-android/releases/tag/5.1.0) which includes a fix for this issue, let us know if you still face any problems with it.

**Comment by ronaldvanduren** [comment]
Sorry for the late reply, it's a different sprint for us and this task was pushed back to the backlog. However, I checked today and the issue is resolved 🎉!
